sodapy/__init__.py

```diff
@@ -76,8 +76,50 @@ def authentication_validation(self, username, password, access_token):
                             " OAuth2.0. Please use only one authentication"
                             " method.")
 
-    def create(self, file_object):
-        raise NotImplementedError()
+    def create(self, name, **kwargs):
+        '''
+        Create a dataset, including the field types. Optionally, specify args such as:
+            description : description of the dataset
+            columns : list of columns (see docs/tests for list structure)
+            category : must exist in /admin/metadata
+            tags : array of tag strings
+            row_identifier : field name of primary key
+        '''
+        public = kwargs.pop("public", False)
+        published = kwargs.pop("published", False)
+        
+        payload = {"name": name}
+        
+        if("row_identifier" in kwargs):
+            payload["metadata"] = {
+                "rowIdentifier": kwargs.pop("row_identifier", None)
+            }
+        
+        payload.update(kwargs)
+        payload = _clear_empty_values(payload)
+        
+        return self._perform_update("post", "/api/views.json", payload)
+    
+    def set_permission(self, resource, permission="private"):
+        '''
+        Set a dataset's permissions to private or public
+        Options are private, public
+        '''
+        params = {
+            "method": "setPermission",
+            "value": "public.read" if permission == "public" else permission
+        }
+        resource = resource.rsplit("/", 1)[-1] # just get the dataset id
+        
+        return self._perform_request("put", "/api/views/" + resource, params=params)
```

Comment:
Yea, you're right. Doing stuff like this is silly. I like the idea of only requiring the user to pass the 4x4 code + content-type extension and then building the URLs in the calls depending on which version of the API we are hitting. That's kind of asking a lot from you though, so I'll merge this first and do the cleanup in a separate commit.